### PR TITLE
Refactor availability code in to its own namespace

### PR DESCRIPTION
**Pull request by freemer**
This is table-setting for GraphQL request throttling.

- created behaviour for rate_limiting/tracker
- moved availability_tracker to rate_limiting/rest_tracker
- moved call_limit to rate_limiting/rest_call_limits
- moved ShopifyAPI helpers functions to rate_limiting/rest
- added tracker_impl arg to Throttled.request

### CODE REVIEW

File: lib/shopify_api/availability/rest.ex [hunk 1]
```diff
@@ -0,0 +1,16 @@
+defmodule ShopifyAPI.Availability.REST do
```
Comment (stewart) [inline lib/shopify_api/availability/rest.ex]: Is `Availability` a better namespacing choice than `RateLimiting`?

Comment (freemer) [reply lib/shopify_api/availability/rest.ex]: We took the naming cue from the existing `AvailabilityTracker`. I have no strong feelings one way or the other, @hez?

Comment (hez) [reply lib/shopify_api/availability/rest.ex]: It was just a cue from the existing code, the age old naming things problem....

- `Availability`
- `RateLimiting`
- `Throttling`

Any other contestants?

**Comment by freemer** [comment]
[ch67324]

File: lib/shopify_api/availability/rest.ex (hunk 1 above)
Comment (freemer) [reply lib/shopify_api/availability/rest.ex]: `RateLimiting` it is!

**Pull request merged**